### DISCUSSION TITLE
Problems connecting to stm32 Devices

### DISCUSSION THREAD

**Discussion by WolleKette** [pycom/pymakr-vsc#297] [General]
I have several microcontroller boards in use. Among others two NUCLEO (L476RG and L432KC) boards with stm32 processors. While it is possible to establish a connection with esp controllers without any problems, for example, it almost never works with stm32 controllers. Visual Studio Code shows me that the connection has been made, but I do not get a REPL prompt.

I do not understand that. With mpremote I always get an immediate connection with all controllers.

**Comment by nastiliano** [comment]
> I have several microcontroller boards in use. Among others two NUCLEO (L476RG and L432KC) boards with stm32 processors. While it is possible to establish a connection with esp controllers without any problems, for example, it almost never works with stm32 controllers. Visual Studio Code shows me that the connection has been made, but I do not get a REPL prompt.
> 
> I do not understand that. With mpremote I always get an immediate connection with all controllers.

Hello,
Sorry but i can't answer your question I'm only a newbie.
I think these discussions are a little bit forgotten. I wrote a post few days ago and nobody answered either.

Best regards.

**Comment by WolleKette** [comment]
Hi, 
thanks a lot for your reply. It's unfortunately the fact that some questions remain unanswered in Internet forums. May be, it's the wrong forum or just no one knows an answer.
In my particular case, however, I don't understand that. The stm controllers are not exotic and Visual Studio is actively developed and used a lot. And that's why I can't believe that only I should have these problems.

Best regards and have a nice day

**Reply by nastiliano** [reply]
Hello,

Anyway, the Pymark extension for VSCode (in my opinion for my experience for last 3 years) is completly experimental and not much updated.
I think I'm going to change to another solution for programming my ESP32.

Best regards.
